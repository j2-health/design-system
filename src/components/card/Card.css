.ant-card:not(.ant-card-bordered) {
  box-shadow: none;

  .ant-card-head {
    border-bottom: none;
  }
}

.ant-card-type-inner {
  background-color: var(--j2-color-bg-layout);
}

.j2-card {
  .ant-card-head {
    border-bottom: none;
    min-height: 40px;
  }

  .ant-tabs .ant-tabs-tab {
    font-size: 14px;
  }
}

.j2-card-default {
  .ant-card-head {
    display: flex;
    justify-content: flex-end;
  }

  .ant-tabs-nav-wrap {
    max-height: 40px;
  }
}

.j2-card-large {
  padding: 0px 32px 32px 32px;

  .ant-card-head {
    /* min-height: 56px; */
    justify-content: flex-end;
    padding: 0;
  }

  .ant-card-body {
    padding: 16px 0px 0px 0px;
  }

  .ant-card-head-wrapper {
    min-height: 56px;
    align-items: flex-end;
  }
}

.j2-card-small {
  .ant-card-head {
    min-height: 38px;

    .ant-card-head-wrapper {
      .ant-card-head-title {
        padding-top: 8px;
        padding-bottom: 8px;
      }
    }
  }
}
