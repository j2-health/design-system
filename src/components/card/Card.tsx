import { Card as AntdCard, CardProps } from 'antd'

// import this file to enable styling defined in there
/* eslint-disable-next-line @typescript-eslint/no-unused-vars */
import './Card.css'

export type Props = CardProps & {
  /** Card title */
  title?: string
  /** Card contents */
  children: React.ReactNode
  /** Loading state */
  loading?: boolean
}

const Card = ({ title, children, loading, ...props }: Props) => {
  return (
    <AntdCard {...props} title={title} loading={loading}>
      {children}
    </AntdCard>
  )
}

export { Card }
