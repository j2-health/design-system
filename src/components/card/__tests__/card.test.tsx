import { render } from '@testing-library/react'
import { Card } from '../Card'

describe('Card', () => {
  it('should render correctly', () => {
    const { container } = render(<Card title="Test">Test</Card>)
    expect(container).toMatchSnapshot()
  })
})
