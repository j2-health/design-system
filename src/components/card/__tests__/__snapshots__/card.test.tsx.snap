// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered j2-card j2-card-undefined css-dev-only-do-not-override-os2fvf"
  >
    <div
      class="ant-card-head"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Test
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      Test
    </div>
  </div>
</div>
`;
